backend/backend/core/agents.py

```diff
@@ -1,22 +1,27 @@
 from langchain_core.prompts import ChatPromptTemplate
+from langgraph.errors import GraphRecursionError
 from langgraph.prebuilt import create_react_agent
 
 from utils.logger import logger
 from .models import ModelRegistry
 from .tools import ToolRegistry
+from .chains import ChainRegistry
 
 
 class EventsAgent(object):
```

Comment:
Consider adding a docstring to the EventsAgent class to describe its purpose and functionality.